packages/react-sdk/src/components/Shortcuts/DuplicateShortcut/DuplicateShortcut.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useCallback } from 'react';
+import { useHotkeys } from 'react-hotkeys-hook';
+import {
+  calculateBoundingRectForElements,
+  useActiveElements,
+  useWhiteboardSlideInstance,
+} from '../../../state';
+import { duplicate } from '../../ElementBar/DuplicateActiveElementButton/DuplicateActiveElementButton';
+import { gridCellSize } from '../../Whiteboard';
+import { HOTKEY_SCOPE_WHITEBOARD } from '../../WhiteboardHotkeysProvider';
+
+export function DuplicateShortcut() {
+  const { activeElementIds } = useActiveElements();
+  const slideInstance = useWhiteboardSlideInstance();
+
+  const handleDuplicate = useCallback(() => {
+    const sortedActiveElementIds =
+      slideInstance.sortElementIds(activeElementIds);
+    const elements = Object.values(
+      slideInstance.getElements(sortedActiveElementIds),
+    );
+    const boundingRect = calculateBoundingRectForElements(elements);
```

Comment:
If extracting the `dublicate` function it would be great to have a function that works on a list of elements. This avoids duplication, e.g. of bounding client rect calculation. This would also simplify the usage in `DuplicateActiveElementButton`.